File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime

class Gigasecond {
    var date: LocalDateTime = LocalDateTime.now()

    constructor(of: LocalDateTime) { date = of.plusSeconds(1000000000) }
    constructor(of: LocalDate) : this(of.atStartOfDay())
}

File: typescript/gigasecond/gigasecond.ts
const MILLIS_IN_GS: number = Math.pow(10, 9) * 1000;

class Gigasecond {
    millisSinceBase: number;

    constructor(base: Date) {
        this.millisSinceBase = base.getTime();
    }

    date(): Date {
        return new Date(this.millisSinceBase + MILLIS_IN_GS);
    }
}

export default Gigasecond
